src/core/net/ip6.hpp

```diff
@@ -340,6 +340,13 @@ class Ip6 : public InstanceLocator, private NonCopyable
      */
     static const char *EcnToString(Ecn aEcn);
 
+    /**
+     * This method returns a reference to the Border Routing counters.
+     *
+     * @returns A reference to the Border Routing counters.
+     */
+    const otBorderRoutingCounters &GetBorderRoutingCounters(void) const { return mBorderRoutingCounters; }
```

Comment:
How about adding support for the counters under a config (maybe `OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE`?)

The `ip6` modules are used on all builds types and products (MTD and SEDs which may be resource limited). This way we limit the overhead (code-size, etc) to BRs (and devices that want to use the counter). 

Looks like it is adding about ~490 bytes.


```
ot-cli-mtd_1.1  main    384168    780   52348   437296
                #8280   384632    780   52380   437792
                +/-     +464      0     +32     +496
```